m4/mpi.m4

```diff
@@ -226,50 +226,44 @@ AS_IF([test "x$MPI_IMPL" != x],
 
 AC_DEFUN([VERSION_MPI], [
 
+  AC_MSG_RESULT([Checking Version of MPI now:])
   AS_IF([test "x$enablempi" != xno],
   [
         dnl check that MPI_VERSION and MPI_SUBVERSION are defined.
         AC_LANG_SAVE
         AC_LANG_CPLUSPLUS
+        dnl first catch undefined and unreasonably high values.
         AC_TRY_LINK([@%:@include <mpi.h>],
-                    [#ifndef MPI_VERSION
-                     #error "MPI-Version undefined."
-                    #endif
+                    [
+                     @%:@ifndef MPI_VERSION
+                     @%:@error "MPI-version seems to be < 1.5."
+                     @%:@endif
+                     @%:@if MPI_VERSION > 3
+                     @%:@error "MPI-version too high. There is some error."
```

Comment:
Are we sure we want to do this?  Seems like it will just break things eventually:
https://www.mpi-forum.org/mpi-40/

Ideally I'd like to print whatever MPI_VERSION we see to make debugging actual "some error" cases, but screaming and dying is too much.

Reply:
So you would prefer to distinguish the different cases and separate these errors? 
I excluded MPI 4 because the standard seems to be still under discussion and so there should be no compatible implementation of it...